src/AzureSignTool/SignCommand.cs

```diff
@@ -355,12 +355,20 @@ public async Task<int> OnExecuteAsync(CommandLineApplication app, IConsole conso
             }
         }
 
-        private static bool IsSigned(string filePath)
+        private static readonly string CodeSigningOid = "1.3.6.1.5.5.7.3.3";
+         
+        public static bool IsSigned(string filePath)
         {
             try
             {
-                _ = X509Certificate.CreateFromSignedFile(filePath);
-                return true;
+                var certificate =  new X509Certificate2(X509Certificate.CreateFromSignedFile(filePath));
```

Comment:
This probably makes it "better" but doesn't really solve the problem, since the problem will still exist if the certificate has a EKU of codeSigning. Maybe we should consider using A Win32 API to determine if it is signed or not.

This is still an improvement over the current situation, so I'll take it.